### ISSUE TITLE
MA0042 shouldn't trigger for `SqlDataReader`

### ISSUE THREAD

**Issue by robv8r** [meziantou/Meziantou.Analyzer#1134]
### Version of the Meziantou.Analyzer NuGet package

3.0.70

### Rule Identifier

MA0042

### Target Framework

net10.0

### C# Language version

C# 14.0

### Description

Given #1132, MA0042 shouldn't be raised for `SqlDataReader.ExecuteReaderAsync()`.

As with `SqlConnection` and `SqlCommand`, `SqlDataReader` calls its base class's `DisposeAsync()` method which calls `Dispose()`.

### Reproduction Steps

Minimal code:

```c#
// 👇 In version 3.0.70, MA0042 is reported even though Dispose() is called in the base class.
using var reader = await command.ExecuteReaderAsync(token).ConfigureAwait(false);
```

**Comment by meziantou** [comment]
The main issue is that the detection is not safe because Sql types are not sealed. But maybe an option similar to Sqlite would be ok.

```c#
SqlCommand command = new MyCustomSqlCommand();
using var reader = command.ExecuteReaderAsync(); // Maybe it returns a class that inherits from SqlDataReader that would benefit from async.
```
